src/views/blockDetails/index.tsx

```diff
@@ -59,15 +65,12 @@ const BlockDetails = () => {
     }
   }, [blockId, blocksData]);
 
-  const latestFinalizedBlock = useStoreChain.use.finalizedBlock?.();
-
-  const handleSetCheck = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-    const target = e.currentTarget.parentNode as HTMLDivElement;
-    const format = target.getAttribute('data-format');
-    target.setAttribute('data-format', format === 'utc' ? 'local' : 'utc');
-
-    setIsChecked(state => !state);
-  }, []);
+  useEffect(() => {
+    if (latestFinalizedBlock && blockData) {
+      console.log(latestFinalizedBlock);
+      latestFinalizedBlock >= blockData?.number ? setIsFinalized(true) : setIsFinalized(false);
```

Comment:
kinda hard to read i think

cosnt blockNumber = blockdata.number || 0
const blockIsFinalized = blockNumber <= latestFinalizedBlock
setIsFinalized(blocIsFinalized)